ci/Makefile

```diff
@@ -2,7 +2,17 @@
 .PHONY: setup-conda-env build-hail-ci push-hail-ci test-locally
 .PHONY: run-service deploy
 
-PROJECT = $(shell gcloud config get-value project)
+PROJECT=${shell gcloud config get-value project}
+
+# Get unique instance name, external_ip, and zone, to avoid hardcoding vars
+# There may be a better way of doing this, but simply ${ARR[N]} did not work
+# 1) Get a unique instance name, either based on passed instance name, passed tag name,
+# or by choosing first listed instance
+# 2) Based on that instance, get EXTERNAL_IP and ZONE, to avoid hardcoding
```

Comment:
We prefer no comments, if the code is too complicated to be understood without commentary we should change the code.

Reply:
I'm not so against comments... There's a worry that they become out of date with the code, but I think we shouldn't forbid them!

Reply:
You have no code comments?

Reply:
My philosophy, which may differ from Cottons (ergo the team's), is that comments should only occur when restructuring the code to obviate the comment is "excessive". Two specific cases in which I think comments are warranted:

 - Bypassing a bug or "feature" of an external dependency. I often accompany these with a URL to a ticket tracking the bug or a third-party discussion (e.g. SO post, GitHub issue) of the feature.

 - Non-idiomatic code specifically addressing a unique performance issue, e.g., mutating to avoid allocation in a core loop.

The latter differs from systematic performance issues like `IndexedSeq` in Scala. Systematic differences from standard style need to be documented in a code style document. We have neglected to maintain the [hail/style-guide.md](https://github.com/hail-is/hail/blob/master/hail/style-guide.md), but that's where this sort of stuff belongs. Preferably with (links to) explanations for the difference. You're right that we haven't done an adequate job of maintaining that information.

I think my reasoning for this philosophy is two-fold: economy and accuracy of comments as well as code simplicity.